### PR TITLE
Support task Element

### PR DESCRIPTION
**Pull request by sknorr**
PR, mainly for the purposes of reviewing this... There must be some reason why you named the commits "first draft", I guess.

### CODE REVIEW

File: suse2013/common/l10n/en.xml [hunk 1]
```diff
@@ -43,6 +43,9 @@
    <l:gentext key="Version" text="Version"/>
 
    <l:gentext key="step.optional" text="(Optional)"/>
+ 
+   <l:gentext key="tasksummary" text="Summary"/>
+   <l:gentext key="taskprerequisites" text="Requirements"/>
```
Comment (sknorr) [inline suse2013/common/l10n/en.xml]: hm, looking at https://github.com/docbook/xslt10-stylesheets/blob/master/gentext/locale/en.xml#L350 , I wonder: are the tasksummary/taskprerequisite/taskrelated elements supposed to have actual titles instead of something autogenerated? does geekodoc allow titles for these elements?

also, (stating the obvious)make sure to add some more languages...

File: suse2013/static/css/style.css [hunk 2]
```diff
@@ -1724,13 +1724,13 @@ p, .simplelist td {
...
-.figure-title .xref, .procedure-title .xref, .orderedlist-title .xref,
+.figure-title .xref, .procedure-title .xref, .task-title .xref, .orderedlist-title .xref,
 .itemizedlist-title .xref, .variablelist-title .xref, .qandadiv-title .xref,
 .title .link, .subtitle .link, .table-title .link, .example-title .link,
-.figure-title .link, .procedure-title .link, .orderedlist-title .link,
+.figure-title .link, .procedure-title .link, .task-title .link, .orderedlist-title .link,
 .itemizedlist-title .link, .variablelist-title .link, .qandadiv-title .link,
 .title .ulink, .subtitle .ulink, .table-title .ulink, .example-title .ulink,
-.figure-title .ulink, .procedure-title .ulink, .orderedlist-title .ulink,
+.figure-title .ulink, .procedure-title .ulink, .task-title .ulink, .orderedlist-title .ulink,
```
Comment (sknorr) [inline suse2013/static/css/style.css]: oh my. i can't even look at this css. totally not your fault but mine. but i almost feel like opening a bug about generating fewer of those class names.

to be clear: your changes are fine and should work. this is more of a note-to-past-self.

File: suse2013/common/labels.xsl [hunk 3]
```diff
@@ -17,5 +17,36 @@
...
+                 select="ancestor::chapter
+                         |ancestor::appendix
+                         |ancestor::article[ancestor::book]"/>
+   <xsl:variable name="prefix">
+     <xsl:if test="count($pchap) &gt; 0">
+       <xsl:apply-templates select="$pchap" mode="label.markup"/>
+     </xsl:if>
+   </xsl:variable>
+   <xsl:choose>
+    <xsl:when test="@label"><xsl:value-of select="@label"/></xsl:when>
```
Comment (sknorr) [inline suse2013/common/labels.xsl]: use two spaces for indentation please. (it seems you're switching between one and two occasionally.)

File: suse2013/xhtml/task.xsl [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+   </xsl:if>
+   <div class="task-contents">
+    <xsl:apply-templates select="$preamble"/>
+   </div>
+  </div>
+ </xsl:template>
+
+ <xsl:template match="tasksummary|taskprerequisites">
+  <div class="{local-name(.)}">
+   <h6 class="{local-name(.)}-title">
```
Comment (sknorr) [inline suse2013/xhtml/task.xsl]: you could hardcode task-title here instead, otherwise you'll have to add lots of extra css to style these extra titles.

File: suse2013/static/css/style.css [hunk 5]
```diff
@@ -2187,7 +2189,8 @@ a:hover .package, a:focus .package, a:active .package {
     display: block;
 }
 
-.procedure-title-wrap + .procedure-contents {
+.procedure-title-wrap + .procedure-contents,
+.task-title-wrap + .task-contents {
```
Comment (sknorr) [inline suse2013/static/css/style.css]: Not directly related to this line, but could find a good place for

```
.task-contents .procedure-contents {
  border-left: none;
}
```

That'll remove the extra line still displayed at the side.

File: suse2013/common/l10n/en.xml (hunk 1 above)
Comment (tomschr) [reply suse2013/common/l10n/en.xml]: > I wonder: are the tasksummary/taskprerequisite/taskrelated elements supposed to have actual titles instead of something autogenerated? Does geekodoc allow titles for these elements?

Currently, no `title` inside of these `task*` elements are allowed in Geekodoc. So yes, that's a feature and not a bug. 😉 

Sure, we could allow titles, but I think in that case we would only get further inconsistencies which would "weaken" our task appearance. At the moment, I can't think of a use case where a title in these `task*` elements is absolutely necessary.

To summarize it, I would like to keep it that way for the time being. Let our writers gain some experiences. If we think we need it, we can add titles in a later version without any compatibility problems.

File: suse2013/static/css/style.css (hunk 2 above)
Comment (tomschr) [reply suse2013/static/css/style.css]: Ok. 👍

File: suse2013/static/css/style.css (hunk 5 above)
Comment (tomschr) [reply suse2013/static/css/style.css]: Ok, see commit 391f98dea50ec

File: suse2013/xhtml/task.xsl (hunk 4 above)
Comment (tomschr) [reply suse2013/xhtml/task.xsl]: Ok, did that.

File: suse2013/common/labels.xsl (hunk 3 above)
Comment (tomschr) [reply suse2013/common/labels.xsl]: Arhg, I was too inconsistent. Could we do it later? Or what procedure do you prefer to change this?

**Comment by tomschr** [comment]
> [...] still lots of positioning issues related to the headlines of "tasksummary" and "taskprerequisites".

I tried to fix them in the CSS, and it looks ok-ish. However, I guess, I did more harm than good. You should review my changes carefully. See commit f4a7b797.

>   why does taskrelated not get a headline in html but does get one in fo? curiously, tasksummary does not get a headline in fo but does get one in html.

This should be fixed now in commit 7fcde21f2, there was an inconsistency in our HTML templates for the `task*` elements.

>    i have a question that relates to geekodoc: why is it not allowed put itemizedlists in taskprerequisites? lists seem a natural fit for prereqs, no?

According to comment https://github.com/openSUSE/geekodoc/issues/1#issuecomment-265167768 we discussed this and we came up with *no* lists. I can't remember why, but I guess, it is an omission.

I've opened openSUSE/geekodoc#25.

File: tests/dapscompare-tests/db5-procedure/xml/test.xml [hunk 6]
```diff
@@ -49,27 +49,195 @@
  </info>
 
  <chapter xml:id="cha.procedure">
-  <title>Procedures</title>
```
Comment (sknorr) [inline tests/dapscompare-tests/db5-procedure/xml/test.xml]: Hm, did I accidentally put the test case stuff into the same commit here on Friday? If so, that was not intended...
